login-workflow/src/new-architecture/components/ChangePasswordDialog/ChangePasswordDialog.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import { defaultPasswordRequirements } from '../../constants';
+import { useAuthContext } from '../../contexts';
+import { useLanguageLocale } from '../../hooks';
+import { ChangePasswordDialogBase } from './ChangePasswordDialogBase';
+import { ChangePasswordDialogProps } from './types';
+
+export const ChangePasswordDialog: React.FC<ChangePasswordDialogProps> = (props) => {
+    const { t } = useLanguageLocale();
+    const passwordRef = useRef(null);
+    const confirmRef = useRef(null);
+    const [passwordInput, setPasswordInput] = useState('');
+    const [confirmInput, setConfirmInput] = useState('');
+    const [currentInput, setCurrentInput] = useState('');
+    const [showErrorDialog, setShowErrorDialog] = useState(false);
+    const [isLoading, setIsLoading] = useState(false);
+    const { actions, navigate, routeConfig } = useAuthContext();
+
+    const passwordRequirements = defaultPasswordRequirements(t);
+
+    const areValidMatchingPasswords = useCallback((): boolean => {
+        for (let i = 0; i < passwordRequirements.length; i++) {
+            if (!new RegExp(passwordRequirements[i].regex).test(passwordInput)) return false;
+        }
+        return confirmInput === passwordInput;
+    }, [passwordRequirements, passwordInput, confirmInput]);
+
+    const updateFields = useCallback(
+        (fields: { password: string; confirm: string }) => {
+            setPasswordInput(fields.password);
+            setConfirmInput(fields.confirm);
+        },
+        [setPasswordInput, setConfirmInput]
+    );
+
+    const currentPasswordChange = (currentPwd: string): void => {
+        setCurrentInput(currentPwd);
+    };
+
+    useEffect(() => {
+        setPasswordInput(areValidMatchingPasswords() ? passwordInput : '');
+    }, [setPasswordInput, passwordInput, confirmInput, areValidMatchingPasswords]);
+
+    const checkPasswords =
+        passwordInput !== '' && confirmInput !== '' && currentInput !== '' && passwordInput === confirmInput;
+
+    const changePasswordSubmit = (): void => {
```

Comment:
This should be async and await on action. Please refer https://github.com/etn-ccis/blui-react-workflows/blob/ddce712207359d5b404dd1fdc7954f44ce680baa/login-workflow/src/new-architecture/screens/EulaScreen/EulaScreen.tsx#L29